src/containers/settings/data/utilities.js

```diff
@@ -0,0 +1,103 @@
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the License for more details.
+ *
+ * You should have received a copy of the License along with this program; if
+ * not, you can obtain one from Tidepool Project at tidepool.org.
+ * == BSD2 LICENSE ==
+ */
+
+import _ from 'lodash';
+
+import * as datetime from '../../../utils/datetime';
+import * as format from '../../../utils/format';
+
+export const DISPLAY_PRESCION_PLACES = 3;
+export const MGDL_UNITS = 'mg/dL';
+export const MMOLL_UNITS = 'mmol/L';
+
+function noData(val) {
+  return val === null || (typeof val === 'undefined');
+}
+
+export function getFormattedTime(startTime) {
+  return datetime.millisecondsAsTimeOfDay(startTime);
+}
+
+export function getScheduleLabel(scheduleName, activeName) {
+  if (scheduleName === activeName) {
+    return `${scheduleName} (Active at upload)`;
+  }
+  return scheduleName;
+}
+
+export function getBasalRate(scheduleData, startTime) {
+  const rate = scheduleData.filter(s => s.start === startTime).map(s => s.rate)[0];
+  if (noData(rate)) {
+    return '';
+  }
+  return format.displayDecimal(rate, DISPLAY_PRESCION_PLACES);
+}
+
+export function getValue(scheduleData, fieldName, startTime) {
+  const val = scheduleData.filter(s => s.start === startTime).map(s => s[fieldName])[0];
+  if (noData(val)) {
+    return '';
+  }
+  return val;
+}
+
+export function getBloodGlucoseValue(scheduleData, fieldName, startTime, units) {
+  const bgValue = getValue(scheduleData, fieldName, startTime);
+  if (noData(bgValue)) {
+    return '';
+  }
+  return format.displayBgValue(bgValue, units);
+}
+
+export function getTotalBasalRates(scheduleData) {
+  const HOUR_IN_MILLISECONDS = 60 * 60 * 1000;
+  const DAY_IN_MILLISECONDS = 86400000;
+
+  let total = 0;
+  for (let i = scheduleData.length - 1; i >= 0; i--) {
+    const start = scheduleData[i].start;
+    let finish = DAY_IN_MILLISECONDS;
+    const next = i + 1;
+    if (next < scheduleData.length) {
+      finish = scheduleData[next].start;
+    }
+    const hrs = (finish - start) / HOUR_IN_MILLISECONDS;
+    total += (scheduleData[i].rate * hrs);
+  }
+  return format.displayDecimal(total, DISPLAY_PRESCION_PLACES);
+}
+
+export function getScheduleNames(settingsData) {
+  return _.keysIn(settingsData);
+}
+
+export function getTimedSchedules(settingsData) {
+  const names = _.map(settingsData, 'name');
+  const schedules = [];
+  for (let i = names.length - 1; i >= 0; i--) {
+    schedules.push({ name: names[i], position: i });
+  }
+  return schedules;
+}
+
+export function getDeviceMeta(settingsData) {
+  return {
+    name: settingsData.deviceId || 'unknown',
+    schedule: settingsData.activeSchedule || 'unknown',
+    uploaded: datetime.formatDisplayDate(settingsData.deviceTime) || 'unknown',
```

Comment:
We don't *ever* use `deviceTime` for display - this is the raw device display timestamp and could be very inaccurate if the user had one of those big mistakes like setting their device to the wrong year.

What should be done is to translate the `time` (this is the bootstrapped value) into a local time using the user's current timezone from their `timePrefs`, which get passed down from blip. There is already a function `getTimezoneFromTimePrefs` in datetime for getting the user's timezone from the `timePrefs` (it's more complex than you might think since there's a `timezoneAware` preference as well). Then you'll need to adjust the `formatDisplayDate` function in datetime to include setting the timezone: `moment.utc(settings.time).tz(timezone).format('MMM Do YYYY')`.